sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -896,22 +969,32 @@ private[joins] object LongHashedRelation {
 
     // Create a mapping of key -> rows
     var numFields = 0
+    val isOriginalInputEmpty: Boolean = !input.hasNext
+    var allNullColumnKeyExistsInOriginalInput: Boolean = false
     while (input.hasNext) {
       val unsafeRow = input.next().asInstanceOf[UnsafeRow]
       numFields = unsafeRow.numFields()
       val rowKey = keyGenerator(unsafeRow)
       if (!rowKey.isNullAt(0)) {
+        // LongToUnsafeRowMap can't insert null key
         val key = rowKey.getLong(0)
         map.append(key, unsafeRow)
+      } else if (!allNullColumnKeyExistsInOriginalInput) {
```

Comment:
Should this check also incorporate `isNullAware`: `else if (isNullAware && ! allNullColumnKeyExistsInOriginalInput)`

Reply:
because LongHashedRelation can't insert null column key, that's why i remove isNullAware from it
in the HashedRelation case class comment, I define isNullAware as

```
If isNullAware is true, UnsafeHashedRelation will store rows with null keys.
```

I will update it if you think it's ok to still put isNullAware inside LongHashedRelation after you have known about this information. ^_^

Reply:
I can't change

```
if (!rowKey.isNullAt(0))

into 

if (isNullAware || !rowKey.isNullAt(0))
because even if isNullAware = true, i still can't insert a null key into LongHashedRelation
```